gremlin-python/src/main/python/gremlin_python/process/graph_traversal.py

```diff
@@ -93,12 +98,30 @@ def with_(self, k, v=None):
     def withRemote(self, remote_connection):
         source = self.get_graph_traversal_source()
         source.traversal_strategies.add_strategies([RemoteStrategy(remote_connection)])
+        self.remote_connection = remote_connection
         return source
 
+    def tx(self):
+        # In order to keep the constructor unchanged within 3.5.x we can try to pop the RemoteConnection out of the
+        # TraversalStrategies. keeping this unchanged will allow user DSLs to not take a break.
+        # This is the same strategy as gremlin-javascript.
+        # TODO: refactor this to be nicer in 3.6.0 when we can take a breaking change
```

Comment:
I used this technique because JavaScript did and I could not come up with a better non-breaking change here. 

It would require a bit of remodelling and it's the kind of thing that could get forgotten, so I think a Jira ticket would be more appropriate.

If you agree I will make one and reference it here.

Reply:
JIRA is fine. i sort of recall the JS approach and that was sorta forced in there a bit so i guess we should make a ticket for that too.

Reply:
https://issues.apache.org/jira/browse/TINKERPOP-2664